prow/cmd/gerrit/main.go

```diff
@@ -90,84 +90,114 @@ func gatherOptions(fs *flag.FlagSet, args ...string) options {
 }
 
 type syncTime struct {
-	val    time.Time
+	val    *client.LastSyncState
 	lock   sync.RWMutex
 	path   string
 	opener io.Opener
 	ctx    context.Context
 }
 
-func (st *syncTime) init() error {
-	fmt.Println(st.val)
+func (st *syncTime) init(hostProjects client.ProjectsFlag) error {
+	logrus.Debug(st.val)
 	st.lock.RLock()
-	zero := st.val.IsZero()
+	zero := st.val == nil
 	st.lock.RUnlock()
 	if !zero {
 		return nil
 	}
 	st.lock.Lock()
 	defer st.lock.Unlock()
-	if !st.val.IsZero() {
+	if st.val != nil {
 		return nil // Someone else set it while we waited for the write lock
 	}
-	unix, err := st.currentInt()
+	state, err := st.currentState()
 	if err != nil {
 		return err
 	}
-	if unix == 0 {
-		st.val = time.Now()
+	if state != nil {
+		st.val = state
 		logrus.Warnf("Reset lastSyncFallback to %v", st.val)
```

Comment:
Does LastSyncState now warrant a String() function? What does this output look like?

Reply:
The default seems good enough? https://play.golang.org/p/yYdpmIKyP3k